src/updateNode.js

```diff
@@ -276,7 +276,12 @@ export default function updateNode (part, node, oldNode, context, forceUpdate, i
     return updateComponentNode(part, node, oldNode, context, forceUpdate, isSvgPart);
   } else if (isTagNode(node)) {
     return updateTagNode(part, node, oldNode, context, forceUpdate, isSvgPart);
-  } else if (isPrimitiveNode(node) && (node !== oldNode || forceUpdate)) {
-    return updateTextNode(part, node, oldNode);
+  } else if (isPrimitiveNode(node)) {
+    const isNodeDifferent = '' + node !== isPrimitiveNode(oldNode)? '' + oldNode: oldNode;
```

Comment:
In most of the time, the type of node will not change. There are only two types which can be rendered. 
number or string. 
Now people having logic to update 1 to '1' will be very uncommon. So adding extra logic for a very niche case will affect the general case.